src/deployment/flavors.md

```diff
@@ -3,41 +3,171 @@ title: Creating flavors for Flutter
 short-title: Flavors
 description: How to create build flavors specific to different release types or development environments.
 ---
+## What are flavors
+Have you ever wondered how to set up different environments in your Flutter app?
+Flavors (known as _build configurations_ in iOS), allow you (the developer) to 
+create separate environments for your app using the same code base. 
+For example, you might have one flavor for your full-fledged production app, 
+another as a limited "free" app, another for testing experimental features, and so on. 
+
+Let's say you want to make both free and paid versions of your Flutter app. 
+You can use flavors to set up both app versions 
+without writing two separate apps. 
+The free version of the app has basic functionality and ads (for example). 
+In contrast, the paid version has basic app functionality, extra features, 
+different styles for paid users, and no ads. 
+
+Another example where you might use flavors is for feature development. 
+If you’ve built a new feature and want to try it out, 
+you could set up a flavor to test out the feature. 
+Your production code remains unaffected 
+until you are ready to deploy your new feature.
+
+Flavors function by letting you define compile time configurations 
+and set parameters that are read at runtime depending on your app needs.
+
+This document guides you through setting up Flutter flavors for iOS and Android. 
+
+## Environment set up
+Prerequisites:
+* Xcode installed
+* An existing project 
+
+To set up flavors in iOS, you’ll define build configurations in Xcode. 
+
+## Creating flavors in iOS
+1. Open your project in Xcode.
+
+2. Select **Runner** > **New Scheme** from the menu to add a new Scheme.  
+* A scheme describes how Xcode runs different actions. 
+  For the purposes of this guide, the example _flavor_ and _scheme_ are 
+  named `development`. 
```

Comment:
I know this would be a lot of work to update the images, but I don't know if `development` is a good example, since that's essentially what "Debug" is. `Debug-production` and `Release-development` don't really make sense.  Free/Paid is a better example, where you would want to debug, QA, and release different versions of a free version and a paid version.

Or if you had an EU and US version of the app, you would want to debug and QA and release each of them.